### PR TITLE
Re-enable concurrency for the pr-tester

### PR DESCRIPTION
**Pull request by M-Davies**
This is an alternative to allowing the jobs to queue up behind one another. At the least the pr-tester jobs will run now.

Closes: #2053 
Signed-off-by: Morgan Davies morgandavies2020@gmail.com

### CODE REVIEW

**Comment by M-Davies** [comment]
Draft until https://github.com/AdoptOpenJDK/openjdk-build/pull/2066 is merged

**Comment by karianna** [comment]
#2066 Has been merged, so you can take this out of draft.  I'll review and we can look to merge post 15 release.

File: pipelines/build/common/config_regeneration.groovy [hunk 1]
```diff
@@ -301,55 +301,50 @@ class Regeneration implements Serializable {
...
 
-                                def pipelineInProgress = queryAPI("${jenkinsBuildRoot}/job/${pipeline}/lastBuild/api/json?pretty=true&depth1")
+                        while (inProgress) {
+                            // Check if pipeline is in progress using api
+                            context.println "[INFO] Checking if ${pipeline} is running..." //i.e. openjdk8-pipeline
 
-                                // If query fails, check to see if the pipeline been run before
-                                if (pipelineInProgress == null) {
-                                    def getPipelineBuilds = queryAPI("${jenkinsBuildRoot}/job/${pipeline}/api/json?pretty=true&depth1")
+                            def pipelineInProgress = queryAPI("${jenkinsBuildRoot}/job/${pipeline}/lastBuild/api/json?pretty=true&depth1")
```
Comment (karianna) [inline pipelines/build/common/config_regeneration.groovy:319]: I was expecting the query to return true or false here, but I see that it's either a result (true) or null (false).  This doesn't need to change, but I'm making the observation for future coding style (converting result to a true/false status helps readers later on).

File: pipelines/build/common/config_regeneration.groovy [hunk 2]
```diff
@@ -301,55 +301,50 @@ class Regeneration implements Serializable {
...
 
-                                    if (getPipelineBuilds.builds == []) {
-                                        context.println "[SUCCESS] ${pipeline} has not been run before. Running regeneration job..."
-                                        inProgress = false
-                                    }
+                            // If query fails, check to see if the pipeline been run before
```
Comment (karianna) [inline pipelines/build/common/config_regeneration.groovy:321]: miss 'has'

File: pipelines/build/common/config_regeneration.groovy [hunk 3]
```diff
@@ -301,55 +301,50 @@ class Regeneration implements Serializable {
...
+                            if (pipelineInProgress == null) {
+                                def getPipelineBuilds = queryAPI("${jenkinsBuildRoot}/job/${pipeline}/api/json?pretty=true&depth1")
 
-                                } else {
-                                    inProgress = pipelineInProgress.building as Boolean
+                                if (getPipelineBuilds.builds == []) {
+                                    context.println "[SUCCESS] ${pipeline} has not been run before. Running regeneration job..."
```
Comment (karianna) [inline pipelines/build/common/config_regeneration.groovy:326]: Is this a case the pipeline has never ever been run before or is there some other time window context here?

File: pipelines/build/common/config_regeneration.groovy [hunk 4]
```diff
@@ -301,55 +301,50 @@ class Regeneration implements Serializable {
...
+                // Get all pipelines
+                def getPipelines = queryAPI("${jenkinsBuildRoot}/api/json?tree=jobs[name]&pretty=true&depth1")
 
-                            while (inProgress) {
-                                // Check if pipeline is in progress using api
-                                context.println "[INFO] Checking if ${pipeline} is running..." //i.e. openjdk8-pipeline
+                // Parse api response to only extract the relevant pipeline
+                getPipelines.jobs.name.each{ pipeline ->
+                    if (pipeline.contains("pipeline") && pipeline.contains(versionNumbers[0])) {
+                        Integer sleepTime = 900
```
[the remark this answers is not in the document]
Comment (karianna) [reply pipelines/build/common/config_regeneration.groovy:312]: What unit is this in as well?

**Review by karianna — CHANGES_REQUESTED** [review]

File: pipelines/build/common/config_regeneration.groovy (hunk 4 above)
[the remark this answers is not in the document]
Comment (M-Davies) [reply pipelines/build/common/config_regeneration.groovy:312]: I haven't changed that line. Nevertheless, it's in seconds. I'll do a quick changes across the board for brevity and magic sake.

I would suggest opening a new issue for parameterising it.

File: pipelines/build/common/config_regeneration.groovy (hunk 1 above)
Comment (M-Davies) [reply pipelines/build/common/config_regeneration.groovy:319]: Again, I haven't changed that line but I can see how that would be better on paper. A future change could be implemented to introduce that

File: pipelines/build/common/config_regeneration.groovy (hunk 3 above)
Comment (M-Davies) [reply pipelines/build/common/config_regeneration.groovy:326]: > Is this a case the pipeline has never ever been run before

That one 😄. The following line fails if the pipeline has never been run as the `building` attribute is not populated
```groovy
Line 331    inProgress = pipelineInProgress.building as Boolean
```

**Review by karianna — APPROVED** [review]

**Review by gdams — APPROVED** [review]

**Pull request merged**